packages/cli/src/index.ts

```diff
@@ -387,33 +392,10 @@ const main = async () => {
       commands.has(targetOrSubcommand);
 
     if (targetPathExists && subcommandExists) {
-      const fileType = lstatSync(targetPath).isDirectory()
-        ? 'subdirectory'
-        : 'file';
-      const plural = targetOrSubcommand + 's';
-      const singular = targetOrSubcommand.endsWith('s')
-        ? targetOrSubcommand.slice(0, -1)
-        : '';
-      let alternative = '';
-      if (commands.has(plural)) {
-        alternative = plural;
-      } else if (commands.has(singular)) {
-        alternative = singular;
-      }
-      console.error(
-        error(
-          `The supplied argument ${param(targetOrSubcommand)} is ambiguous.` +
-            `\nIf you wish to deploy the ${fileType} ${param(
-              targetOrSubcommand
-            )}, first run "cd ${targetOrSubcommand}". ` +
-            (alternative
-              ? `\nIf you wish to use the subcommand ${param(
-                  targetOrSubcommand
-                )}, use ${param(alternative)} instead.`
-              : '')
-        )
+      output.warn(
```

Comment:
The case where this logs is narrower than it was before. Because the global option changes the process directory, this code doesn't detect a conflicting subdirectory. Because the option `--cwd` is used, there's not command, either.

To make this trigger, we now have to use a command and the `--cwd` option where value of that option points to a directory that itself includes a subdirectory with the same name as the command. In that case, I think you are right that we don't want to warn that they might have meant a different `--cwd` option.

I just added a check for the arg being used.